### PR TITLE
Friendships v2

### PR DESCRIPTION
**Pull request by Victorgonbu**
## Hi there, in this feature I worked on the following features implementing friendships with 2 rows for mutual friendship.

- Add friend (creates 1 row)
- Cancel friend request (delete the 1 row created)
- Decline friend request (delete the 1 row created by another user)
- Accept friend request (updates the first row created and creates another row for mutual friendship)
- Delete friend (deletes the 2 rows created)

Friendship model, helpers, request, and features tested with RSpec.

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -14,19 +14,22 @@ def create
...
-    @friendship = current_user.inverse_friendships.find_by(user_id: params[:id])
+    @friendship = current_user.friendship_requests.find_by(user_id: params[:id])
     if @friendship
       @friendship.update(confirmed: true)
+      Friendship.create(user_id: params[:user_id], friend_id: params[:id], confirmed: true)
       redirect_back(fallback_location: root_path, notice: 'Friend request accepted')
     else
       redirect_back(fallback_location: root_path, alert: 'Unable to send friend request')
     end
   end
```
Comment (kubilaycaglayan) [inline app/controllers/friendships_controller.rb]: image

- [x]  [OPTIONAL] Follow the MVC best practices.

Green ones are OK to be in the controllers. 
For the red ones, you can create a class method in the Friendship or in the User, to put that logic. And then you can call that method itself right after the `if`.

That way, you will be following the best practices of the MVC architecture.

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
#### Hello Team 👋
#### Status: Changes Required :recycle: 

Great job so far! 👏👋
You have done the most important part of this milestone. The double row feature!
BRAVO :blush: :+1: 
Let's add more tests to make your project better 😊⏬

### Testing the Application

- [x] Add more unit tests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

:bulb: Currently, we only have unit tests for the Friendship model. Please extend your testing coverage for all the other models.

## That's it, you can request another code review request after making the changes ♻️😇

With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

File: app/controllers/friendships_controller.rb (hunk 1 above)
Comment (kubilaycaglayan) [reply app/controllers/friendships_controller.rb]: This is about keeping the controllers thin and models thick.

Comment (Victorgonbu) [reply app/controllers/friendships_controller.rb]: Thank you very much for your review

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

Your tests are exceptional! Nailed it 💯 !

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Comment by Victorgonbu** [comment]
Thank you very much @tirthajyoti-ghosh

**Pull request merged**